packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -169,6 +169,30 @@ export default class CallDiagnosticLatencies extends WebexPlugin {
     );
   }
 
+  /**
+   * Getting catalog time from U2C service
+   * @returns - latency
+   */
+  public getU2CTime() {
+    const preAuth =
+      this.getDiffBetweenTimestamps(
+        'internal.get.u2c.request-preauth',
+        'internal.get.u2c.response-preauth'
+      ) || 0;
```

Comment:
suggestion: It may be better to check that the diff is a positive number, since a negative number (which could happen if there is an in-flight catalog request whilst this method is called) would make the overall timing lower than it should be. Same applies to the postauth version